src/handlers/system.rs

```diff
@@ -1,75 +1,112 @@
+    cached: u64,
+    swap_total: u64,
+    swap_free: u64,
 }
 
-#[get("/sys/disk")]
-pub async fn get_disk() -> (Status, String) {
-    match disk_info() {
-        Ok(disk) => return (Status::Ok, format!("{}/{}", disk.free, disk.total)),
-        Err(_) => return (Status::BadRequest, "failed to get disk".to_string())
+// implement from for MemInfoDef
+impl From<MemInfo> for MemInfoDef {
+    fn from(mem_info: MemInfo) -> Self {
+        MemInfoDef {
+            total: mem_info.total,
+            free: mem_info.free,
+            avail: mem_info.avail,
+            buffers: mem_info.buffers,
+            cached: mem_info.cached,
+            swap_total: mem_info.swap_total,
+            swap_free: mem_info.swap_free,
+        }
     }
 }
 
-#[get("/sys/cpuspeed")]
-pub async fn get_cpu_speed() -> (Status, String) {
-    match cpu_speed() {
-        Ok(cpuspeed) => return (Status::Ok, cpuspeed.to_string()),
-        Err(_) => return (Status::BadRequest, "failed to get cpu speed".to_string())
+pub async fn get_ram() -> Json<MemInfoDef> {
+    Json(mem_info().expect("Failed to get memory info").into())
+}
+
+// Since DiskInfo is not serializable, we need to create a new struct that is serializable.
+#[derive(Serialize, Deserialize)]
+pub struct DiskInfoDef {
+    total: u64,
+    free: u64,
+}
+
+// implement from for DiskInfoDef
+impl From<DiskInfo> for DiskInfoDef {
+    fn from(disk_info: DiskInfo) -> Self {
+        DiskInfoDef {
+            total: disk_info.total,
+            free: disk_info.free,
+        }
     }
 }
-/// DOES NOT WORK IN WSL
-#[get("/sys/cpuinfo")]
-pub async fn get_cpu_info() -> (Status, String) {
-    // TODO: get CPU info without extra dependencies
-    (Status::Ok, "testcpu".to_string())
-    // match cpuid::identify() {
-    //     Ok(cpuinfo) => return (Status::Ok, format!("{} {}", cpuinfo.vendor, cpuinfo.codename)),
-    //     Err(_) => return (Status::BadRequest, "failed to get cpu info".to_string())
-    // }
+
+pub async fn get_disk() -> Json<DiskInfoDef> {
+    Json(disk_info().expect("Failed to get disk info").into())
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CPUInfo {
+    pub cpu_vendor: String,
+    pub cpu_speed: u64,
+    pub cpu_load: f32,
 }
-/// This handler will always take 1s+ to respond
-#[get("/sys/cpuutil")]
-pub async fn get_utilization() -> (Status, String) {
+
+pub async fn get_cpu() -> Json<CPUInfo> {
     let sys = System::new();
-    match sys.cpu_load_aggregate() {
-        Ok(load) => {
-            thread::sleep(Duration::from_secs(1));
-            return (Status::Ok, load.done().unwrap().user.to_string())
-        },
-        Err(_) => return (Status::BadRequest, "failed to get cpu info".to_string())
-    }
+    let cpu_aggregate = sys.cpu_load_aggregate().expect("Failed to get cpu load");
+    thread::sleep(Duration::from_secs(1));
+    let cpu_load = cpu_aggregate.done().expect("Failed to get cpu load");
+
+    Json(CPUInfo {
+        cpu_vendor: "TODO".to_string(),
```

Comment:
extra info is sent on handler that is periodic pulled

Reply:
removed

Reply:
just remember to implement it later